src/services/iiif-parser.js

```diff
@@ -118,14 +138,20 @@ export function parseStructureToJSON(manifest, duration, canvasIndex = 0) {
         if (range) {
           const childCanvases = range.getCanvasIds();
           let structItem = {};
-          if (childCanvases.length > 0) {
+          // Check if the Canvas for this item exists in the Manifest
+          const canvasIsInManifest = isCanvasInManifest(childCanvases[0], manifest);
+          if (childCanvases.length > 0 && canvasIsInManifest) {
             const { start, end } = getMediaFragment(childCanvases[0], duration);
             structItem = {
               label: getLabelValue(i.label),
               type: "span",
               begin: smUtils.toHHmmss(start),
               end: smUtils.toHHmmss(end),
             };
+            if (i.items) {
```

Comment:
As far as I can tell this handles the case where Ranges with Canvas items also contain other child Ranges.  Are there any tests which test this block?

Reply:
Yes, I believe so. I don't think it has its own unit test, but it covered by new unit test for nested structures. I will collapse this logic add tests as needed..